packages/compass-preferences-model/src/provider.ts

```diff
@@ -1,18 +1,68 @@
 import { createContext, useContext } from 'react';
-import type { PreferencesAccess } from './';
-import { defaultPreferencesInstance } from '.';
+import type {
+  AllPreferences,
+  PreferenceStateInformation,
+  PreferencesAccess,
+  UserConfigurablePreferences,
+} from './';
+import { getDefaultsForAllPreferences } from './preferences-schema';
 export { usePreference, withPreferences } from './react';
 export { capMaxTimeMSAtPreferenceLimit } from './maxtimems';
 
+export class ReadOnlyPreferenceService implements PreferencesAccess {
+  private allPreferences: AllPreferences;
+  constructor(preferencesOverrides?: Partial<AllPreferences>) {
+    this.allPreferences = {
+      ...getDefaultsForAllPreferences(),
+      ...preferencesOverrides,
+    };
+  }
+
+  savePreferences() {
+    return Promise.resolve(this.allPreferences);
+  }
+
+  refreshPreferences() {
+    return Promise.resolve(this.allPreferences);
+  }
+
+  getPreferences() {
+    return this.allPreferences;
+  }
+
+  ensureDefaultConfigurableUserPreferences() {
+    return Promise.resolve();
+  }
+
+  getConfigurableUserPreferences() {
+    return Promise.resolve({} as UserConfigurablePreferences);
+  }
```

Comment:
While the naming is still a bit confusing here (this is more like `AllowedToBeConfiguredByTheUserInUIIfServiceAllowsIt`), I think we can copy the same logic that the electron-main version of the preferences does and filter them based on the `ui` flag of the preferences configuration

Reply:
Good point, thank you for this. I will have that.
With `AllowedToBeConfiguredByTheUserInUIIfServiceAllowsIt` do you mean a `Partial< UserConfigurablePreferences>` given that not all preferences are user configurable in all the environments?

Reply:
No, not really. Based on the implementation, `ConfigurableUserPreferences` in preferences seems to refer to preferences that we theoretically allow user to configure (as opposed to those that can only be set through global configuration file in Compass), not preferences that can be actually changed by calling `savePreferences` as we initially assumed from the name. So`ConfigurableUserPreferences` will be exactly the same for compass and data explorer, at least it's safe to assume this for now